clients/azuredevopsrepo/work_items.go

```diff
@@ -0,0 +1,175 @@
+
+type (
+	fnQueryWorkItems func(
+		ctx context.Context,
+		args workitemtracking.QueryByWiqlArgs,
+	) (*workitemtracking.WorkItemQueryResult, error)
+
+	fnGetWorkItems func(
+		ctx context.Context,
+		args workitemtracking.GetWorkItemsArgs,
+	) (*[]workitemtracking.WorkItem, error)
+
+	fnGetWorkItemComments func(
+		ctx context.Context,
+		args workitemtracking.GetCommentsArgs,
+	) (*workitemtracking.CommentList, error)
+)
+
+type workItemsHandler struct {
+	ctx                 context.Context
+	repourl             *Repo
+	once                *sync.Once
+	errSetup            error
+	workItemsClient     workitemtracking.Client
+	queryWorkItems      fnQueryWorkItems
+	getWorkItems        fnGetWorkItems
+	getWorkItemComments fnGetWorkItemComments
+	issues              []clients.Issue
+}
+
+func (w *workItemsHandler) init(ctx context.Context, repourl *Repo) {
+	w.ctx = ctx
+	w.errSetup = nil
+	w.once = new(sync.Once)
+	w.repourl = repourl
+	w.queryWorkItems = w.workItemsClient.QueryByWiql
+	w.getWorkItems = w.workItemsClient.GetWorkItems
+	w.getWorkItemComments = w.workItemsClient.GetComments
+}
+
+func (w *workItemsHandler) setup() error {
+	w.once.Do(func() {
+		// Fetch URI, CreatedAt, Author, and Issue comments
+		wiql := `
+			SELECT [System.Id]
+			FROM WorkItems
+			WHERE [System.TeamProject] = @project
+			ORDER BY [System.Id] DESC
+		`
+		workItems, err := w.queryWorkItems(w.ctx, workitemtracking.QueryByWiqlArgs{
+			Project: &w.repourl.project,
+			Wiql: &workitemtracking.Wiql{
+				Query: &wiql,
+			},
+		})
+		if err != nil {
+			w.errSetup = fmt.Errorf("error getting work items: %w", err)
+			return
+		}
+
+		ids := make([]int, 0, len(*workItems.WorkItems))
+		for _, wi := range *workItems.WorkItems {
+			ids = append(ids, *wi.Id)
+		}
+
+		// Get details for each work item
+		workItemDetails, err := w.getWorkItems(w.ctx, workitemtracking.GetWorkItemsArgs{
+			Ids: &ids,
+		})
+		if err != nil {
+			w.errSetup = fmt.Errorf("error getting work item details: %w", err)
+			return
+		}
+
+		// Get comments for each work item
+		for i := range *workItemDetails {
+			wi := &(*workItemDetails)[i]
+
+			createdBy, ok := (*wi.Fields)["System.CreatedBy"].(map[string]interface{})
+			if !ok {
+				w.errSetup = errSystemCreatedByFieldNotMap
+				return
+			}
+			createdDate, ok := (*wi.Fields)["System.CreatedDate"].(string)
+			if !ok {
+				w.errSetup = errSystemCreatedDateFieldNotString
+				return
+			}
+			parsedTime, err := time.Parse(time.RFC3339, createdDate)
+			if err != nil {
+				w.errSetup = fmt.Errorf("error parsing created date: %w", err)
+				return
+			}
+			// There is not currently an official API to get user permissions in Azure DevOps
+			// so we will default to RepoAssociationMember for all users.
+			repoAssociation := clients.RepoAssociationMember
+
+			issue := clients.Issue{
+				URI:               wi.Url,
+				CreatedAt:         &parsedTime,
+				Author:            &clients.User{Login: createdBy["uniqueName"].(string)},
```

Comment:
you check all your other type conversions, did you mean to here?

Reply:
Added a type assertion for `uniqueName` too.